tests/baselines/reference/excessPropertyCheckWithUnions.errors.txt

```diff
@@ -71,33 +71,36 @@ tests/cases/compiler/excessPropertyCheckWithUnions.ts(114,63): error TS2322: Typ
     // no error for ambiguous tag, even when it could satisfy both constituents at once
     amb = { tag: "A", x: "hi" }
     amb = { tag: "A", y: 12 }
+                      ~~~~~
+!!! error TS2322: Type '{ tag: "A"; y: number; }' is not assignable to type 'Ambiguous'.
+!!! error TS2322:   Object literal may only specify known properties, and 'y' does not exist in type '{ tag: "A"; x: string; }'.
     amb = { tag: "A", x: "hi", y: 12 }
+                               ~~~~~
+!!! error TS2322: Type '{ tag: "A"; x: string; y: number; }' is not assignable to type 'Ambiguous'.
+!!! error TS2322:   Object literal may only specify known properties, and 'y' does not exist in type '{ tag: "A"; x: string; }'.

```

Comment:
These are...ambiguous! Unclear if they should error.

Reply:
Wait, why should they be errors? Particularly `amb = { tag: "A", y: 12 }` seems super wrong. But this one too seems like excess property checking is a little overzealous. For non-discriminated unions of objects, excess property checking only complains about properties non present in any constituent. Here, I would think the same rule would apply after we have filtered out the B and C constituents.